hikari/impl/event_manager.py

```diff
@@ -56,6 +57,21 @@ def _fixed_size_nonce() -> str:
     return base64.b64encode(head + tail).decode("ascii")
 
 
+async def _request_guild_members(
+    shard: gateway_shard.GatewayShard,
+    guild: guilds.PartialGuild,
+    *,
+    include_presences: bool,
+    nonce: str,
+) -> None:
+    try:
+        await shard.request_guild_members(guild, include_presences=include_presences, nonce=nonce)
+
+    # Ignore errors raised by a shard shutting down
+    except errors.ComponentStateConflictError:
```

Comment:
Do we want to make more precise errors for each component that derive from `ComponentStateConflictError`?

Reply:
what's the use case for that

Reply:
Will probably clarify all these excepts, as it would be something like `errors.ShardStateConflictError`

Reply:
Eh, at least under the current interface these errors don't really ever bobble up like that and unless there's a reason to want to differentiate between them at runtime (can't really think of one myself) the error's message should already be enough to clarify where it's from. What we could do with however is added documentation on places where it might be raised since right now we're lacking on that